### PR TITLE
ci: limit releases to only occur on relevant dependency updates

### PR DESCRIPTION
**Pull request by acazacu**
Limit releases triggered by development dependency updates to only occur when relevant packages are updated.

### CODE REVIEW

File: .github/workflows/continuous-delivery.yaml [hunk 1]
```diff
@@ -39,12 +39,19 @@ jobs:
           commit-assets: |
             ./dist
           node-module: true
+          release-rules:
+            '[ { "release": "patch", "type": "build" }, { "release": "patch",
+            "type": "chore", "scope": "deps" }, { "release": "patch", "type":
+            "chore", "scope": "deps-dev", "message": "*typescript*" }, {
+            "release": "patch", "type": "ci" }, { "release": "patch", "type":
+            "docs" }, { "release": "patch", "type": "improvement" }, {
+            "release": "patch", "type": "refactor" }, ]'
```
Comment (merlinnot) [inline .github/workflows/continuous-delivery.yaml:48]: Can we somehow format it? I can't read it without copy-pasting to an editor.

File: .github/workflows/continuous-delivery.yaml [hunk 2]
```diff
@@ -39,12 +39,19 @@ jobs:
...
       - if: steps.release.outputs.released == 'true'
         name: Authenticate
         uses: actions/checkout@v2
         with:
-          fetch-depth: 1
           persist-credentials: true
+          ref: master
```
Comment (merlinnot) [inline .github/workflows/continuous-delivery.yaml:54]: I have trouble understanding why is this change necessary.

File: .github/workflows/continuous-delivery.yaml (hunk 1 above)
Comment (mad-it) [inline .github/workflows/continuous-delivery.yaml:48]: Could you make each rule fit on a single line making it a bit more readable?

Comment (acazacu) [reply .github/workflows/continuous-delivery.yaml:48]: I did, however, it was overridden by Prettier :(

Comment (merlinnot) [reply .github/workflows/continuous-delivery.yaml:48]: Btw., it's not a valid JSON!

File: .github/workflows/continuous-delivery.yaml [hunk 3]
```diff
@@ -39,12 +39,19 @@ jobs:
           commit-assets: |
             ./dist
           node-module: true
+          release-rules:
+            '[ { "release": "patch", "type": "build" }, { "release": "patch",
+            "type": "chore", "scope": "deps" }, { "release": "patch", "type":
+            "chore", "scope": "deps-dev", "message": "*typescript*" }, {
```
Comment (merlinnot) [inline .github/workflows/continuous-delivery.yaml:45]: Match on `typescript` is too loose.

File: .github/workflows/continuous-delivery.yaml (hunk 2 above)
Comment (acazacu) [reply .github/workflows/continuous-delivery.yaml:54]: This bled here from the other PR, which I would like merged (or closed) first .

File: .github/workflows/continuous-delivery.yaml (hunk 1 above)
Comment (merlinnot) [reply .github/workflows/continuous-delivery.yaml:48]: It's a YAML file, YAML has multiple ways to format strings and you'll definitely find one that you can format yourself, e.g. `: |` syntax.

File: .github/workflows/continuous-delivery.yaml [hunk 4]
```diff
@@ -39,6 +39,16 @@ jobs:
           commit-assets: |
             ./dist
           node-module: true
+          release-rules: |
+            [
+              { "release": "patch", "type": "build" },
+              { "release": "patch", "type": "chore", "scope": "deps" },
+              { "release": "patch", "type": "chore", "scope": "deps-dev", "message": "bump typescript from*" },
+              { "release": "patch", "type": "ci" },
+              { "release": "patch", "type": "docs" },
```
Comment (mad-it) [inline .github/workflows/continuous-delivery.yaml:48]: Should docs trigger a release?

File: .github/workflows/continuous-delivery.yaml [hunk 5]
```diff
@@ -39,6 +39,16 @@ jobs:
           commit-assets: |
             ./dist
           node-module: true
+          release-rules: |
+            [
+              { "release": "patch", "type": "build" },
+              { "release": "patch", "type": "chore", "scope": "deps" },
+              { "release": "patch", "type": "chore", "scope": "deps-dev", "message": "bump typescript from*" },
```
Comment (mad-it) [inline .github/workflows/continuous-delivery.yaml:46]: At this moment simple chores like `chore: cleanup depedencies` dont get a release.

File: .github/workflows/continuous-delivery.yaml (hunk 4 above)
Comment (acazacu) [reply .github/workflows/continuous-delivery.yaml]: Nope! Resolved!

I'm thinking now, should we also whitelist `@zeit/ncc` and `@ridedott/eslint-config`? They could both change the compiled output, albeit, only aesthetically and not functionally.

Comment (mad-it) [reply .github/workflows/continuous-delivery.yaml]: `eslint` does not automatically change your code. It requires a human intervention to. As for ncc 🤷‍♂️

File: .github/workflows/continuous-delivery.yaml (hunk 5 above)
Comment (acazacu) [reply .github/workflows/continuous-delivery.yaml:46]: Fixed!

File: .github/workflows/continuous-delivery.yaml (hunk 4 above)
Comment (acazacu) [reply .github/workflows/continuous-delivery.yaml]: Alright, I'm leaving them out. We will add them later on if we notice they are being missed.

Comment (merlinnot) [reply .github/workflows/continuous-delivery.yaml]: @mad-it @acazacu ncc _does_ change the output. This should trigger a release.

File: .github/workflows/continuous-delivery.yaml (hunk 2 above)
Comment (acazacu) [reply .github/workflows/continuous-delivery.yaml]: Removed!

File: .github/workflows/continuous-delivery.yaml [hunk 6]
```diff
@@ -39,6 +39,16 @@ jobs:
...
+              { "release": "patch", "type": "chore" },
```
Comment (merlinnot) [inline .github/workflows/continuous-delivery.yaml:47]: I don't see how would that exclude `deps-dev` from being released. Did you test it or is there some documentation that would explain such behavior? I don't see how would that work.

File: .github/workflows/continuous-delivery.yaml [hunk 7]
```diff
@@ -39,6 +39,16 @@ jobs:
...
+              { "release": "patch", "type": "ci" },
+              { "release": "patch", "type": "improvement" },
+              { "release": "patch", "type": "refactor" }
+            ]
```
Comment (merlinnot) [inline .github/workflows/continuous-delivery.yaml:51]: We should add `ncc` to the list.

File: .github/workflows/continuous-delivery.yaml (hunk 6 above)
Comment (acazacu) [reply .github/workflows/continuous-delivery.yaml:49]: Changed, this should do the trick. I think!

There are no explicit docs around how release rules work (is order important?), only examples to paint some very specific use cases.

File: .github/workflows/continuous-delivery.yaml (hunk 5 above)
Comment (acazacu) [reply .github/workflows/continuous-delivery.yaml:46]: Actually, now it should be fixed! 😬
